EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp

```diff
@@ -330,6 +330,54 @@ Signal::Type Signal::GetTypeFromString(const std::string& type)
 	return Signal::TYPE_UNKNOWN;
 }
 
+void Signal::Save(pugi::xml_node& signals)
+{
+	auto signal = signals.append_child("signal");
+	signal.append_attribute("s").set_value(s_);
+	signal.append_attribute("t").set_value(t_);
+	signal.append_attribute("id").set_value(id_);
+	signal.append_attribute("name").set_value(name_.c_str());
+	if(dynamic_)
+		signal.append_attribute("dynamic").set_value("true");
+	else if(!dynamic_)
+		signal.append_attribute("dynamic").set_value("false");
+
+	switch (orientation_)
+	{
+	case RoadObject::Orientation::NEGATIVE :
+		signal.append_attribute("orientation").set_value("-");
+		break;
+	case RoadObject::Orientation::POSITIVE :
+		signal.append_attribute("orientation").set_value("+");
+		break;
+	case RoadObject::Orientation::NONE :
+		signal.append_attribute("orientation").set_value("none");
+		break;
+	default:
+		break;
+	}
+	signal.append_attribute("zOffset").set_value(z_offset_);
+	signal.append_attribute("country").set_value(country_.c_str());
+	signal.append_attribute("type").set_value(type_);
+	//object.append_attribute("subtype").set_value() TODO:
```

Comment:
remove

Reply:
These kind of commented out stuff I left since there are no parser for these attributes. Will add TODO mark aswell.